app/js/components/change_member_role.ts

```diff
@@ -2,39 +2,48 @@ import * as $ from 'jquery';
 import addFlashMessage from './addFlashMessage';
 
 $(() => {
-  const selects = document.querySelectorAll('select[data-url]');
   const flashUl = document.querySelector('.teams__flashMessages') as HTMLUListElement;
+  const changeRoleLinks = document.querySelectorAll('.teams__changeMemberRole');
+  changeRoleLinks.forEach(item => {
+    item.addEventListener('mousedown', (event) => {
+      const target = event.target as any;
+      const roleName = target.getAttribute('role');
+      if (target.closest('div').getAttribute('data-url')) {
+        const parent = target.closest('td') as HTMLTableDataCellElement;
+        const name = (parent?.previousElementSibling?.previousElementSibling?.textContent || '').trim();
```

Comment:
I'd rewrite this slightly for better legibility:

`const name = (parent?.previousElementSibling?.previousElementSibling?.textContent).trim()  || ''`  That makes it clearer that you either use the textcontent of that element when trimmed, or an empty string (which doesn't need to be trimmed). 

Aside from that, using the `.closest` method with a CSS selector would be better than traversing via the `previousElementSibling` as that's very closely tied to the HTML implementation.

Reply:
it has been rewritten to your suggestion